src/GHC/Util.hs

```diff
@@ -325,3 +330,33 @@ wToStr :: Outputable a => W a -> String
 wToStr (W e) = showPpr baseDynFlags e
 instance Outputable a => Eq (W a) where (==) a b = wToStr a == wToStr b
 instance Outputable a => Ord (W a) where compare = compare `on` wToStr
+
+-- GHC parse trees don't contain pragmas. We work around this with
+-- (nasty) parsing of comments.
+
+-- Pragmas. Comments not associated with a span in the annotations
+-- that have the form @{-# ...#-}@.
+pragmas :: ApiAnns -> [(Located AnnotationComment, String)]
+pragmas anns =
+  [ (c, s) |
+      c@(L _ (AnnBlockComment comm)) <- fromMaybe [] $ Map.lookup noSrcSpan (snd anns)
+    , let body = trimCommentDelims comm
+    , "#" `isPrefixOf` body && "#" `isSuffixOf` body
+    , let s = trim $ take (length body - 2) (drop 1 body)
+  ]
+
+-- Flags. The first element of the pair is the (located) annotation
+-- comment that sets the flags enumerated in the second element of the
+-- pair.
+flags :: ApiAnns -> [(Located AnnotationComment, [String])]
+flags anns =
+  [(c, opts) | (c, s) <- pragmas anns, "OPTIONS_GHC" `isPrefixOf` s
```

Comment:
I think `OPTIONS` alone without `OPTIONS_GHC` probably also wants to be detected as a pragma. Pragmas are also case insensitive, so `options` and `optIONs` work just fine.

Reply:
I see. The next PR in the queue does something about `OPTIONS`.

Reply:
I remember now. There was a twitter thread about `oPTionS` just recently!

Reply:
✔️